Source/RunActivity/Viewer3D/Sky.cs

```diff
@@ -135,15 +135,28 @@ public void PrepareFrame(RenderFrame frame, ElapsedTime elapsedTime)
 
 
             // Current solar and lunar position are calculated by interpolation in the lookup arrays.
+            // The arrays have intervals of 1200 secs or 20 mins.
             // Using the Lerp() function, so need to calculate the in-between differential
-            float diff = (float)(Viewer.Simulator.ClockTime - oldClockTime) / 1200;
+            float diff = GetCelestialDiff();
             // The rest of this increments/decrements the array indices and checks for overshoot/undershoot.
             while (Viewer.Simulator.ClockTime >= (oldClockTime + 1200)) // Plus key, or normal forward in time; <CSComment> better so in case of fast forward
             {
+                //step1++;
+                //step2++;
+                //oldClockTime = oldClockTime + 1200;
+                //diff = (float)(Viewer.Simulator.ClockTime - oldClockTime) / 1200;
+                //if (step2 >= maxSteps) // Midnight.
+                //{
+                //    step2 = 0;
+                //}
+                //if (step1 >= maxSteps) // Midnight.
+                //{
+                //    step1 = 0;
+                //}
```

Comment:
These lines should be deleted instead of commented out, the deleted lines will be in the git history anyway.
```suggestion
```